tests/io/v3/test_credentials.py

```diff
@@ -0,0 +1,218 @@
+'''
+Testing for credentials API
+'''
+
+import os
+
+import responses
+
+CREDENTIALS_BASE_URL = r'https://cloud.tenable.com/api/v3/credentials'
+TYPES_RESPONSE = {
+    'credentials': [
+        {
+            "id": "API Gateway",
+            "category": "API Gateway",
+            "default_expand": False,
+            "types": [
+                {
+                    "id": "IBM DataPower Gateway",
+                    "name": "IBM DataPower Gateway",
+                    "max": 1,
+                    "configuration": [
+                        {
+                            "type": "file",
+                            "name": "Client Certificate",
+                            "hint": "PEM formatted certificate.",
+                            "id": "datapower_client_cert"
+                        },
+                        {
+                            "type": "file",
+                            "name": "Client Certificate Private Key",
+                            "hint": "PEM formatted certificate.",
+                            "id": "datapower_private_key"
+                        },
+                        {
+                            "type": "password",
+                            "name": "Client Certificate "
+                                    "Private Key Passphrase",
+                            "id": "datapower_private_key_passphrase"
+                        },
+                        {
+                            "type": "text",
+                            "name": "Custom Header Key",
+                            "id": "datapower_custom_header_key"
+                        },
+                        {
+                            "type": "text",
+                            "name": "Custom Header Value",
+                            "id": "datapower_custom_header_value"
+                        },
+                        {
+                            "type": "checkbox",
+                            "name": "Enable for Hashicorp Vault",
+                            "required": True,
+                            "default": "yes",
+                            "id": "datapower_enable_hashicorp"
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+}
+DETAILS_RESPONSE = {
+    "name": "Windows devices (Headquarters)",
+    "description": "Use for scans of Windows devices located at headquarters.",
+    "category": {
+        "id": "Host",
+        "name": "Host",
+    },
+    "type": {
+        "id": "Windows",
+        "name": "Windows",
+    },
+    "ad_hoc": False,
+    "user_permissions": 64,
+    "settings": {
+        "domain": "",
+        "username": "user@example.com",
+        "auth_method": "Password",
+        "password": "********"
```

Comment:
I'm thinking we want to drop the `password` property here. A hidden password isn't really helpful, and we certainly do not want to return the password in clear text (which I know isn't what we are doing now).